assets/js/modules/analytics/components/setup/SetupMain.js

```diff
@@ -64,6 +66,22 @@ export default function SetupMain( { finishSetup } ) {
 		select( MODULES_ANALYTICS ).getSetupFlowMode()
 	);
 
+	const isTagManagerAvailable = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleAvailable( 'tagmanager' )
+	);
+	// Preloading the `live-container-version` call is necessary to display
+	// the loading state until the data is fetched from the server.
+	// This call is being made in the child component ExistingGTMPropertyNotice
+	// to ensure that the loading state is displayed before the data is available.
+	useSelect(
+		( select ) =>
+			isTagManagerAvailable &&
+			select( MODULES_TAGMANAGER ).getSingleAnalyticsPropertyID()
+	);
+	const gtmContainersResolved = useSelect( ( select ) =>
+		select( MODULES_TAGMANAGER ).hasFinishedLoadingContainers()
+	);
+
```

Comment:
Good thought here.

Since this component conditionally renders `<SetupForm />` which then unconditionally renders `<ExistingGTMPropertyNotice />` within its children, how about we move this pre-loading logic to `<SetupForm />`? Of course, that would mean rendering an additional `<ProgressBar />` in `<SetupForm />`, but would ensure we don't pre-load containers unnecessarily.

Reply:
Good thought, thanks @nfmohit 👍